app/helpers/flyover_comments/comments_helper.rb

```diff
@@ -2,13 +2,25 @@ module FlyoverComments
   module CommentsHelper
 
     def flyover_comment_content(comment)
-      if comment.deleted_at.nil?
-        comment.content
-      else
+      if comment.deleted_at?
         I18n.t('flyover_comments.comments.comment_deleted_text', deleted_time_stamp: comment.deleted_at.to_s(:normal))
+      elsif show_flagged_text?(comment)
+        I18n.t('flyover_comments.comments.comment_flagged_text')
+      else
+        comment.content
       end
     end
 
+    def show_flagged_text?(comment)
+      try(:hide_flagged_comments?) && comment.flags.not_reviewed.exists? && !membership_can_view_flagged_comment?(comment)
+    end
+
+    def membership_can_view_flagged_comment?(comment)
+      # if there isn't a current_membership, or the membership doesn't have a method called
+      # can_view_flagged_comment?, then this will return false
+      current_membership&.try(:can_view_flagged_comment?, comment)
+    end
+
```

Comment:
I need to merge this so I'm going to make the changes, but FYI:

1. I'd generally consider using `try` a code smell, and especially in this case where we're trying to access methods that have no definition within this gem (e.g. `hide_flagged_comments?`). 

2. This gem shouldn't have knowledge of the membership concept